lua/curl/cache.lua

```diff
@@ -21,7 +22,10 @@ local curl_cache_dir = function(custom_dir)
 		cache_dir = cache_dir / custom_dir ---@type Path
 	end
 
-	cache_dir:mkdir({ parents = true, exists_ok = true })
+	if vim.fn.mkdir(cache_dir:absolute(), "p") ~= 1 then
+		notify.error("create directory error: " .. cache_dir:absolute())
```

Comment:
This message goes to the user, we should perhaps formalize it a little bit:

"Error creating collection: could not create directory " .. cache_dir:absolute()